### PR TITLE
chore: Add support for split pagination modifier

### PR DESCRIPTION
**Pull request by teneightfive**
### What changed

Add a modifier to the pagination component to allow it to be split across the container element.

Also includes a tidy up to the examples. Examples should only show visual changes. Custom options are document in the yaml files and should be tested within creating visual examples that will be rendered in the component explorer.

### Why did it change

This allows pagination to be displayed on either side of the container
element and will be used to pagination results in a collection.

## Screenshots

<kbd>image</kbd>

[**DEMO**](https://booksecuremove-pr-1202.herokuapp.com/components/pagination)

#### Automated testing

- [x] Added unit tests to cover changes

#### Manual testing

Has been tested in the following browsers:

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### CODE REVIEW

File: common/components/pagination/pagination.yaml [hunk 1]
```diff
@@ -93,38 +86,33 @@ examples:
           text: "two"
         - href: "/page-3"
           text: "three"
-  - name: with items with class
```
Comment (teneightfive) [inline common/components/pagination/pagination.yaml:96]: These were removed from the examples as each example is rendered in the components explorer. These two variations had no visual difference so that's why they've been removed from here and are now tested separately within the module test file.

File: common/components/pagination/template.test.js [hunk 2]
```diff
@@ -184,60 +184,72 @@ describe('Pagination component', function () {
       expect($itemLink.attr('href')).to.equal('/page-3')
     })
   })
-})
 
-context('with items with class', function () {
-  let $, $component, items
+  context('with items with class', function () {
```
Comment (teneightfive) [inline common/components/pagination/template.test.js:188]: The indentation of these tests was previously incorrect, they were sat outside the containing block that describes this component. That's why this change is showing some whitespace changes.

**Review by cwrw — APPROVED** [review]
